src/request/sign-transaction/SignTransaction.js

```diff
@@ -1,71 +1,116 @@
+        const $senderDetails = (this.$el.querySelector('#details > .sender'));
         // eslint-disable-next-line no-new
-        new Identicon(transaction.sender.toUserFriendlyAddress(), $senderIdenticon);
-
-        const $senderAddresses = ($sender.querySelectorAll('.address > .chunk'));
-        /** @type {string[]} */
-        const senderAddressChunks = (
-            transaction.sender
-                .toUserFriendlyAddress()
-                .replace(/[+ ]/g, '').match(/.{4}/g)
+        new AddressInfo(
+            $senderDetails,
+            {
+                userFriendlyAddress: transaction.sender.toUserFriendlyAddress(),
+                label: request.senderLabel || 'Unnamed Contact',
+                imageUrl: null,
+                accountLabel: request.keyLabel || null,
+            },
         );
-        $senderAddresses.forEach(($el, x) => {
-            $el.textContent = senderAddressChunks[x];
-        });
 
-        if (request.senderLabel) {
-            /** @type {HTMLElement} */
-            const $senderLabel = ($sender.querySelector('.label'));
-            $senderLabel.textContent = request.senderLabel;
-            $senderLabel.classList.remove('display-none');
+        /** @type {HTMLLinkElement} */
+        const $recipient = (this.$el.querySelector('.account.recipient'));
+
+        /** @type {HTMLLinkElement} */
+        const $recipientDetails = (this.$el.querySelector('#details > .recipient'));
+        switch (request.layout) {
+        case SignTransactionApi.Layouts.checkout:
+            this._recipientAddressInfo = new AddressInfo(
+                $recipient,
+                {
+                    userFriendlyAddress: transaction.recipient.toUserFriendlyAddress(),
+                    label: /** @type {string} */ (request.shopOrigin).split('://')[1],
+                    imageUrl: request.shopLogoUrl || null,
+                    accountLabel: null,
+                },
+            );
+            // eslint-disable-next-line no-new
+            new AddressInfo(
+                $recipientDetails,
+                {
+                    userFriendlyAddress: transaction.recipient.toUserFriendlyAddress(),
+                    label: /** @type {string} */ (request.shopOrigin).split('://')[1],
+                    imageUrl: request.shopLogoUrl || null,
+                    accountLabel: null,
+                },
+            );
+            break;
+        case SignTransactionApi.Layouts.standard:
+        default:
+            this._recipientAddressInfo = new AddressInfo(
+                $recipient,
+                {
+                    userFriendlyAddress: transaction.recipient.toUserFriendlyAddress(),
+                    label: request.recipientLabel || transaction.recipient.toUserFriendlyAddress(),
+                    imageUrl: null,
+                    accountLabel: null,
+                },
+            );
+            // eslint-disable-next-line no-new
+            new AddressInfo(
```

Comment:
As discussed, I'd like a "stateful" component better here, one that knows which place it gets rendered in (what its 
_purpose_ is). As such, it would be able to determine itself which parts of itself to display and which to hide and what data to take as fallback for other unavailable props. It would also contain the logic and style of the component within it, as opposed to external overwriting CSS from parent contexts.

Using a further config param, would enable this and enable you to reduce the number of config object definitions from 6 to 2, once for sender, once for recipient.